ui/src/app/views/cohort-list/component.ts

```diff
@@ -71,12 +71,37 @@ export class CohortListComponent implements OnInit, OnDestroy {
   }
 
   public deleteCohort(cohort: Cohort): void {
+    if (this.awaitingReview || !this.writePermission) {
```

Comment:
Is this condition needed, since were checking this above for the class.disabled? Seems if the button is disabled then this function should never get called in this component?

Reply:
Class.disabled is actually just for display purposes, it doesn't block a click function (unfortunately). As far as I can tell there is no way to block click...

Reply:
I believe [disabled]="condition" will grey out the button and block click

Reply:
Trying now. (I had some problems with that before, but it may be a built in clarity directive, and thus didn't work for other clicks?)